### ISSUE TITLE
Headers under :includes: should actually be included at the top

### ISSUE THREAD

**Issue by Xunnamius** [ThrowTheSwitch/Unity#265]
This is in reference to `auto/generate_test_runner.rb`.

For example, I use CException with some custom configurations `#define`-d in a header file, but even if I include `cexception_configured.h` in the yml settings file, the generator will place it below `CException.h` and I'll get redefinition problems.

```C
/*=======Automagically Detected Files To Include=====*/
#include "unity.h"
#include <setjmp.h>
#include <stdio.h>
#include "CException.h"
#include "cexception_configured.h"
#include "bitmask.h"
```

In my own fork, I've made it so that files under the `:includes:` array in the settings yml file are included above the others, including `CException.h` in this example with the cexception plugin enabled. If I have the time later today, I'll try for a pull request.

Excellent project. Thanks!

**Comment by mvandervoord** [comment]
The best solution is to handle it the way CMock doesn, which is to add specific options like:
    :includes_h_pre_orig_header 
    :includes_h_post_orig_header 
    :includes_c_pre_header 
    :includes_c_post_header

We've had a similar request in the generate_module script and I was planning to port these options over there too... the :includes default would just become an alias for :includes_c_post_header, which is the most common request.

**Comment by mvandervoord** [comment]
Fixed in commit 689610b
